### PR TITLE
conjunct path

### PR DESCRIPTION
**Pull request by nevermore3**

### CODE REVIEW

File: src/executor/test/ConjunctPathTest.cpp [hunk 1]
```diff
@@ -15,221 +15,700 @@ namespace nebula {
...
-            // 1->3
-            DataSet ds1;
-            ds1.colNames = {kVid, "edge"};
-            {
-                Row row;
-                row.values = {"1", Value::kEmpty};
-                ds1.rows.emplace_back(std::move(row));
-            }
-            qctx_->ectx()->setResult("forward1", ResultBuilder().value(ds1).finish());
+static bool comparePath(Row& row1, Row& row2) {
```
Comment (yixinglu) [inline src/executor/test/ConjunctPathTest.cpp:18]: please format this file and fix the indent.

File: src/executor/test/ConjunctPathTest.cpp [hunk 2]
```diff
@@ -758,5 +1232,194 @@ TEST_F(ConjunctPathTest, AllPathsFourStepsPath) {
...
+        // 1->5->7->10->11->12, 1->6->7->10->11->12
+        for (auto i = 5; i < 7; i++) {
+            Row row;
+            Path path;
+            path.src = Vertex("1", {});
+            path.steps.emplace_back(Step(Vertex(folly::to<std::string>(i), {}), 1, "edge1", 0, {}));
+            path.steps.emplace_back(Step(Vertex("7", {}), 1, "edge1", 0, {}));
+            path.steps.emplace_back(Step(Vertex("10", {}), 1, "edge1", 0, {}));
+            path.steps.emplace_back(Step(Vertex("11", {}), 1, "edge1", 0, {}));
+            path.steps.emplace_back(Step(Vertex("12", {}), 1, "edge1", 0, {}));
```
Comment (yixinglu) [inline src/executor/test/ConjunctPathTest.cpp:1350]: Why not to provide a path build function to extract this logics for all tests, like:

```cpp
Path createPath(const string &src, const std::vector<string>& steps) {
  Path path;
  path.src = src;
  for (auto& step : steps) {
    path.steps.emplace_back(Step(Vertex(step, {)), 1, "edge1", 0, {});
  }
  return path;
}

// usage
auto path = createPath("1", {folly::to<string>(i), "7", "10", "11"});
```

File: src/executor/algo/ConjunctPathExecutor.cpp [hunk 3]
```diff
@@ -18,6 +18,8 @@ folly::Future<Status> ConjunctPathExecutor::execute() {
             return bfsShortestPath();
         case ConjunctPath::PathKind::kAllPaths:
             return allPaths();
+        case ConjunctPath::PathKind::kFloyd:
+            return conjunctPath();
```
Comment (CPWstatic) [inline src/executor/algo/ConjunctPathExecutor.cpp:22]: Naming `conjunctPath ` accordingly with Floyd.

File: src/executor/algo/ConjunctPathExecutor.cpp [hunk 4]
```diff
@@ -176,59 +179,100 @@ folly::Future<Status> ConjunctPathExecutor::conjunctPath() {
...
     }
 
-    auto latest = rHist.back().iter();
-    findPath(latest.get(), table, ds);
     return finish(ResultBuilder().value(Value(std::move(ds))).finish());
 }
 
-bool ConjunctPathExecutor::findPath(Iterator* iter,
-                                    std::multimap<Value, const Path*>& table,
+Status ConjunctPathExecutor::unionPath(const List& forwardPaths,
```
Comment (CPWstatic) [inline src/executor/algo/ConjunctPathExecutor.cpp:217]: `union` seems not a proper name.

File: src/executor/query/DataCollectExecutor.cpp [hunk 5]
```diff
@@ -183,5 +184,69 @@ Status DataCollectExecutor::collectAllPaths(const std::vector<std::string>& vars
     result_.setDataSet(std::move(ds));
     return Status::OK();
 }
+
+Status DataCollectExecutor::collectMultiplePairShortestPath(const std::vector<std::string>& vars) {
```
Comment (CPWstatic) [inline src/executor/query/DataCollectExecutor.cpp:188]: Please optimize it according to the offline discussion.

**Review by yixinglu — APPROVED** [review]

**Review by CPWstatic — APPROVED** [review]

**Pull request merged**